gcp_variant_transforms/beam_io/vcfio.py

```diff
@@ -720,7 +728,9 @@ def _store_header_lines(self, header_lines):
     def _create_generator(self):
       header_processed = False
       for text_line in self._text_lines:
-        if not header_processed and self._header_lines:
+        if (not self._representative_headers_given and
```

Comment:
You'd need to parse the last line of the header from each file separately (it contains the sample IDs, which is unique per file). I think the easiest approach is to modify `store_header_lines` to modify self._header_lines accordingly using the `representative_header_lines` variable. You'd then leave this logic unchanged.